paasta_tools/cli/cmds/local_run.py

```diff
@@ -492,6 +500,24 @@ def get_local_run_environment_vars(instance_config, port0, framework):
     return env
 
 
+def check_if_port_free(port):
```

Comment:
can you make this function just return True or False, and move the exception handling outside this function? otherwise *every* caller of this function will have to sys.exit if the port isn't free, which isn't ideal.